src/object_gui.cpp

```diff
@@ -543,9 +543,38 @@ class BuildObjectWindow : public Window {
 
 	void OnPlaceObject(Point pt, TileIndex tile) override
 	{
-		ObjectClass *objclass = ObjectClass::Get(_selected_object_class);
-		Command<CMD_BUILD_OBJECT>::Post(STR_ERROR_CAN_T_BUILD_OBJECT, CcPlaySound_CONSTRUCTION_OTHER,
-				tile, objclass->GetSpec(_selected_object_index)->Index(), _selected_object_view);
+		const ObjectSpec *spec = ObjectClass::Get(_selected_object_class)->GetSpec(_selected_object_index);
+
+		if (spec->size == 0x11) {
```

Comment:
Not sure if there already is a proper named constant for this, but it should either be one or at the very least get a comment explaining the magic number.